lib/discourse_code_review/github_repo.rb

```diff
@@ -213,19 +213,24 @@ def git(command, backup_command: nil, raise_error: true, warn: true)
       end
 
       Dir.chdir(path) do
-        result = `git #{command}`.strip
-        if $?.exitstatus != 0
-          if backup_command
-            result = `git #{backup_command}`.strip
+        begin
+          result = Discourse::Utils.execute_command('git', *command).strip
+        rescue
```

Comment:
A bit uneasy with a blanket eat all rescue, this means we are losing info. especially in cases where raise_error: true is set

can you specifically catch the right error, and use that info for the logger warn?

Reply:
I can fix the warning, but the execute_shell just raises a string, so that needs a coordinated fix.